syncopy/io/_read_nwb.py

```diff
@@ -0,0 +1,245 @@
+        raise SPYValueError(lgl, varname=ttlVals[0].name, actual=act)
+
+    # Use provided TTL data to initialize `EventData` object
+    evtData = None
+    if len(ttlVals) > 0:
+        msg = "Creating separate EventData object for embedded TTL pulse data..."
+        SPYInfo(msg)
+        evtData = EventData(dimord=EventData._defaultDimord)
+        h5evt = h5py.File(evtData.filename, mode="w")
+        evtDset = h5evt.create_dataset("data", dtype=np.result_type(*ttlDtypes),
+                                       shape=(ttlVals[0].data.size, 3))
+        evtDset[:, 0] = ((ttlChans[0].timestamps[()] - tStarts[0]) / ttlChans[0].timestamps__resolution).astype(np.intp)
+        evtDset[:, 1] = ttlVals[0].data[()]
+        evtDset[:, 2] = ttlChans[0].data[()]
+        evtData.data = evtDset
+        evtData.samplerate = float(1 / ttlChans[0].timestamps__resolution)
+        if hasTrials:
+            evtData.trialdefinition = trl
+
+    # Allocate `AnalogData` object and use generated HDF5 file-name to manually
+    # allocate a target dataset for reading the NWB data
+    angData = AnalogData(dimord=AnalogData._defaultDimord)
+    angShape = [None, None]
+    angShape[angData._defaultDimord.index("time")] = nSamples
+    angShape[angData._defaultDimord.index("channel")] = nChannels
+    h5ang = h5py.File(angData.filename, mode="w")
+    angDset = h5ang.create_dataset("data", dtype=np.result_type(*dTypes), shape=angShape)
+
+    # Compute actually available memory (divide by 2 since we're working with an add'l tmp array)
+    pbarDesc = "Reading data in blocks of {} GB".format(round(memuse / 1000, 2))
+    memuse *= 1024**2 / 2
+    chanCounter = 0
+
+    # Process analog time series data and save stuff block by block (if necessary)
+    pbar = tqdm(angSeries, position=0)
+    for acqValue in pbar:
+
+        # Show dataset name in progress bar label
+        pbar.set_description("Loading {} from disk".format(acqValue.name))
+
+        # Given memory cap, compute how many data blocks can be grabbed per swipe
+        nSamp = int(memuse / (np.prod(angDset.shape[1:]) * angDset.dtype.itemsize))
+        rem = int(angDset.shape[0] % nSamp)
+        nBlocks = [nSamp] * int(angDset.shape[0] // nSamp) + [rem] * int(rem > 0)
+
+        # If channel-specific gains are set, load them now
+        if acqValue.channel_conversion is not None:
+            gains = acqValue.channel_conversion[()]
+
+        # Write data block-wise to `angDset` (use `del` to wipe blocks from memory)
+        # Use 'unsafe' casting to allow `tmp` array conversion int -> float
+        endChan = chanCounter + acqValue.data.shape[1]
+        for m, M in enumerate(tqdm(nBlocks, desc=pbarDesc, position=1, leave=False)):
+            tmp = acqValue.data[m * nSamp: m * nSamp + M, :]
+            if acqValue.channel_conversion is not None:
+                np.multiply(tmp, gains, out=tmp, casting="unsafe")
+            angDset[m * nSamp: m * nSamp + M, chanCounter : endChan] = tmp
```

Comment:
Now I see why you don't like users calling our DataClass objects datasets :) But this semantical weight we have to carry as devs, for the user the data is what they load/see and operate on..